app/scripts/services/gateways-service.js

```diff
@@ -47,6 +47,7 @@ sc.service('Gateways', function($q, $analytics, session, StellarNetwork, rpStell
 
   Gateways.remove = function(gateway) {
     gateway.status = "removing";
+    walletGateways()[gateway.domain].status = "removing";
```

Comment:
just made this one change to the service, actually this just matches what goes on in the 'add' method. Without this the tests would not pass (unless I would already set the status of the gateway in the session to 'removing')